File: Java/Spark/spark-edu/src/main/scala/com/bistel/edu/TransformMap.scala
package com.bistel.edu

import org.apache.spark.SparkConf
import org.apache.spark.SparkContext;
import org.apache.spark.SparkContext._;

object TransformMap extends App{

  val conf = new SparkConf().setAppName("TransformMap").setMaster("local[*]")
  val sc = new SparkContext(conf)
  
  val rdd = sc.parallelize(1 to 5)
  val a = rdd.map(_*2).collect
  val b = rdd.map(x => (x.toString, (x*x*x)))
  
  println("A============================")
  a.foreach(println)
  println("B============================")
  b.foreach(println)
  
  val c = rdd.filter(_%2!=0).collect
  
  println("C============================")
  c.foreach(println)
  
  
  def pr(text:String, xs:Any*){
    println()
    printf("[Result of ]" + text + " ----> ")
    println(xs)
    //println("")
    xs.foreach(println)
  }
 
  pr("rdd.map(_*2)", rdd.map(_*2).collect)
}

File: Java/Spark/SparkSample_JAVA/src/main/java/com/spark_java/sample/TransformMap.java
package com.spark_java.sample;

//import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
//import java.util.regex.Pattern;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
//import org.apache.spark.api.java.function.FlatMapFunction;
//import org.apache.spark.api.java.function.Function;
//import org.apache.spark.sql.DataFrame;
//import org.apache.spark.sql.SQLContext;
//import org.apache.spark.storage.StorageLevel;

//import scala.Tuple2;
//import bean.table01;

 
public class TransformMap {
	
//	private static final Pattern SPACE = Pattern.compile(" ");

	public static void main(String[] args) throws Exception {
		
		JavaSparkContext ctx = new JavaSparkContext(new SparkConf().setAppName("WordCount").setMaster("local[*]"));
		
		List<Integer> data = Arrays.asList(1, 2, 3, 4, 5);
		JavaRDD<Integer> distData = ctx.parallelize(data);
	    //HiveContext
	    //JavaRDD<String> lines = ctx.textFile("C:/spark-1.5.0-bin-hadoop2.6/README.md", 1);
	    //temp.map(f)
		JavaRDD<Integer> lineLengths = distData.map(s -> s * 2);
		System.out.println(lineLengths.collect().toString());
		
		JavaRDD<Object> lineLengthsa = distData.map(x -> (x + ",(" +x*x*x+ ")"));
		System.out.println(lineLengthsa.collect().toString());
	    //JavaRDD<table01> temp = temps.map(null);
	   
		ctx.stop();
	    ctx.close();
	}
	  
}